src/pudl/io_managers.py

```diff
@@ -426,8 +426,9 @@ def __init__(
 
         super().__init__(base_dir, db_name, md, timeout)
 
-        existing_schema_context = MigrationContext.configure(self.engine.connect())
-        metadata_diff = compare_metadata(existing_schema_context, self.md)
+        with self.engine.connect() as conn:
```

Comment:
Found an un-closed connection when testing - figured I'd clean that up while I'm here.